File: solutions/julia/secret-handshake/secret-handshake.jl
function secret_handshake(code)
    actions = String[]

    code = bitstring(UInt8(code))

    if code[8] == '1' push!(actions, "wink") end
    if code[7] == '1' push!(actions, "double blink") end
    if code[6] == '1' push!(actions, "close your eyes") end
    if code[5] == '1' push!(actions, "jump")  end

    if code[4] == '1' actions = reverse(actions) end

    return actions

end

File: solutions/haskell/secret-handshake/src/SecretHandshake.hs
module SecretHandshake (handshake) where

import Data.Bits as Bits

handshake :: Int -> [String]
handshake n =
    if n .&. 16 == 16 then (reverse moves) else moves
    where
        moves = filter (not . null) [
                    if n .&. 1 == 1 then "wink"            else [],
                    if n .&. 2 == 2 then "double blink"    else [],
                    if n .&. 4 == 4 then "close your eyes" else [],
                    if n .&. 8 == 8 then "jump"            else []
                ]